oval-schemas/independent-definitions-schema.xsd

```diff
@@ -968,6 +968,171 @@
+                  <xsd:appinfo>
+                        <oval:element_mapping>
+                              <oval:test>shellcommand_test</oval:test>
+                              <oval:object>shellcommand_object</oval:object>
+                              <oval:state>shellcommand_state</oval:state>
+                              <oval:item target_namespace="http://oval.mitre.org/XMLSchema/oval-system-characteristics-5#independent">shellcommand_item</oval:item>
+                        </oval:element_mapping>
+                  </xsd:appinfo>
+                  <xsd:appinfo>
+                        <sch:pattern id="ind-def_shellcommand_tst">
+                              <sch:rule context="ind-def:shellcommand_test/ind-def:object">
+                                    <sch:assert test="@object_ref=ancestor::oval-def:oval_definitions/oval-def:objects/ind-def:shellcommand_object/@id"><sch:value-of select="../@id"/> - the object child element of a shellcommand_test must reference a shellcommand_object</sch:assert>
+                              </sch:rule>
+                              <sch:rule context="ind-def:shellcommand_test/ind-def:state">
+                                    <sch:assert test="@state_ref=ancestor::oval-def:oval_definitions/oval-def:states/ind-def:shellcommand_state/@id"><sch:value-of select="../@id"/> - the state child element of a shellcommand_test must reference a shellcommand_state</sch:assert>
+                              </sch:rule>
+                        </sch:pattern>
+                  </xsd:appinfo>
+            </xsd:annotation>
+            <xsd:complexType>
+                  <xsd:complexContent>
+                        <xsd:extension base="oval-def:TestType">
+                              <xsd:sequence>
+                                    <xsd:element name="object" type="oval-def:ObjectRefType" />
+                                    <xsd:element name="state" type="oval-def:StateRefType" minOccurs="0" maxOccurs="unbounded"/>
+                              </xsd:sequence>
+                        </xsd:extension>
+                  </xsd:complexContent>
+            </xsd:complexType>
+      </xsd:element>
+      
+      <xsd:element name="shellcommand_object" substitutionGroup="oval-def:object">
+            <xsd:annotation>
+                  <xsd:documentation>The shellcommand_object is used by a shellcommand_test to define a shell to use (e.g. sh, bash, ksh, etc.), a command (or shell script) to be run, and a pattern to filter result lines. The default shell is bash.  Each object extends the standard ObjectType as defined in the oval-definitions-schema and one should refer to the ObjectType description for more information. The common set element allows complex objects to be created using filters and set logic.
```

Comment:
I think the default shell should be:
1) cmd.exe on Windows
2) /bin/sh on Unix/Linux/macOS

Reply:
@solind how do we go about having 2 different default values?  I'm obviously ignorant on this, I assumed in XML you could only have one default, which is partly why we originally wanted to have this be 2 different tests, one for Windows and one for UNIX.

Reply:
@vanderpol You can't. If the element is optional, however, and there is no default value set ... then we can document a meaning for that, which would be to set it to the platform-specific value.